pkg/pquerier/frames.go

```diff
@@ -31,8 +22,16 @@ type frameIterator struct {
 }
 
 // create new frame set iterator, frame iter has a SeriesSet interface (for Prometheus) plus columnar interfaces
-func NewFrameIterator(ctx *selectQueryContext) *frameIterator {
-	return &frameIterator{ctx: ctx, columnNum: ctx.totalColumns, setIndex: 0, seriesIndex: -1}
+func NewFrameIterator(ctx *selectQueryContext) (*frameIterator, error) {
+	if !ctx.isRawQuery() {
+		for _, f := range ctx.frameList {
+			if err := f.finishAllColumns(); err != nil {
+				return nil, err
```

Comment:
Shouldn't it also says which dataframe failed? (i.e. wrap the error with this information)